tests/p2p_connection_test.rs

```diff
@@ -15,119 +15,45 @@ mod p2p_connection_test {
+        let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
 
+        // Test successful p2p connection.
+        {
+            sleep(Duration::from_secs(100));
             assert_eq!(node.get_node_info().num_peers, 1);
-            let peers = setup::nigiri::list_peers(NodeInstance::NigiriLnd).unwrap();
+            let peers = setup::nigiri::list_peers(NodeInstance::LspdLnd).unwrap();
             assert!(peers.contains(&node.get_node_info().node_pubkey.to_hex()));
         }
 
-        // test failing p2p connection
+        // Test disconnect when LSP is down.
         {
-            let lsp_info = setup::nigiri::query_node_info(NodeInstance::NigiriLnd).unwrap();
-            let lsp_node = NodeAddress {
-                pub_key: lsp_info.pub_key,
-                host: FAULTY_ADDR.to_string(),
-            };
-
-            let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
+            // Let's shutdown LSPD LND.
+            setup::nigiri::pause_lspd();
+            sleep(Duration::from_secs(1));
 
             assert_eq!(node.get_node_info().num_peers, 0);
-            assert!(setup::nigiri::list_peers(NodeInstance::NigiriLnd)
-                .unwrap()
-                .is_empty());
         }
 
-        // test start node while lsp is down
+        // Test reconnect when LSP is back.
         {
-            let lsp_info = setup::nigiri::query_node_info(NodeInstance::NigiriLnd).unwrap();
-            let lsp_node = NodeAddress {
-                pub_key: lsp_info.pub_key,
-                host: NIGIRI_LND_ADDR.to_string(),
-            };
-
-            // Let's shutdown LND, while we leave Esplora running (test should not affect chain sync)
-            setup::nigiri::pause();
-            setup::nigiri::resume_without_ln();
-
-            let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
-
-            assert_eq!(node.get_node_info().num_peers, 0);
-
             // Now let's start Nigiri with LND again
-            setup::nigiri::pause();
-            setup::nigiri::resume();
-
-            // Wait for the LDK to connect to the LSP (which should now succeed)
-            sleep(Duration::from_millis(1500));
-
-            assert_eq!(node.get_node_info().num_peers, 1);
-            let peers = setup::nigiri::list_peers(NodeInstance::NigiriLnd).unwrap();
-            assert!(peers.contains(&node.get_node_info().node_pubkey.to_hex()));
-        }
-
-        // test flaky p2p connection
-        {
-            let lsp_info = setup::nigiri::query_node_info(NodeInstance::NigiriLnd).unwrap();
-            let lsp_node = NodeAddress {
-                pub_key: lsp_info.pub_key,
-                host: NIGIRI_LND_ADDR.to_string(),
-            };
-            let node = NodeHandle::new(lsp_node.clone()).start().unwrap();
-
+            setup::nigiri::start_lspd();
+            setup::nigiri::wait_for_healthy_lspd();
+            sleep(Duration::from_secs(10));
```

Comment:
Same as in the other comment. If we run the reconnect task more frequently in tests (e.g. 1 sec), we can decrease this sleep to 1-2 secs.